STYLE.md

```diff
@@ -131,15 +131,172 @@ if status is not pebble.ServiceStatus.ACTIVE:
+
+Example:
+
+- Avoid: "For reference information about the API, see [API and clients] and [API]." (It's hard to tell the difference of these two.)
+- Prefer: "For an explanation of API access levels, see [API and clients]. For the full API reference, see [API reference]." (Clear)
+
+### Don't over-promote
+
+Avoid overstatement: Only use adjectives like "comprehensive", "powerful", or "robust" when the feature truly meets the description.
+
+Example:
+
+- Avoid: "Pebble provides a comprehensive health check feature"
+- Prefer: "Pebble provides a health check feature"
+
+Let features speak for themselves, describe actual capabilities and use measurable terms when possible (for example, "supports 3 types of checks" instead of "versatile checking")
+
+Future-proofing: Qualify with "currently" when describing evolving features. For example: "Currently, the only supported log target type is Grafana Loki".
+
+### Avoid implementation-specific terminology
+
+Focus on behaviour, not internal implementation: Describe what it does rather than how a particular implementation works.
+
+Example:
+
+- Avoid: "`Change.Err` will be non-empty if the change had an error." (Code-specific)
+- Prefer: "The `err` field in the response will contain an error message if the operation failed." (General)
+
+Exception: If documenting a client library (for example, Go/Python SDKs), implementation details are appropriate.
+
+### Articles
+
+Choose between the "a" or "the" articles carefully.
+
+- Avoid: "The format of the duration string is a sequence of decimal numbers"
+- Prefer: "The format of a duration string is a sequence of decimal numbers." (Describing a generic parameter, not a specific one.)
+
+- Avoid: "Restart the service when the health check fails."
+- Prefer: "Restart a service when the health check fails." (When describing a generic behaviour, no specific service is implied.)
+
+
+### Code blocks
+
+- Consistency: the styles of code blocks and terminal output samples should be the same, at least within the same document.
+- Preferred style: Use `{code-block}` when showing files. Use `{terminal}` (`.rst` style) when showing commands or terminal output. Avoid using `{code-block}` for commands or terminal output, unless it's consistent with the existing content in the same document.
+- Highlighting: Use `:emphasize-lines: 8-10` with `{code-block}` for highlighting lines in files when necessary. Don't add an inconsistent `{code-block}` just so that you can highlight lines in commands or terminal output. Instead, use `{terminal}` along with helpful words and comments.
+
+### YAML
+
+- Use quotes for strings: This is especially important if a string contains special characters or starts with a number.
+- Indentation: Always use spaces and be consistent with the number of spaces throughout the same file (usually, this should be two or four).
```

Comment:
I'd be in favour of us picking one (probably 2?) and going with that rather than just saying to be consistent.

```suggestion
- Indentation: Always use spaces and be consistent with the number of spaces throughout the same file (two, unless the file already uses a different number).
```